x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/swimlane/swimlane_connectors.tsx

```diff
@@ -0,0 +1,385 @@
+    [editActionConfig, mappings]
+  );
+  //
+  // const editProperties = useCallback(
+  //   (key: string, value: unknown) => {
+  //     if()
+  //   },
+  //   [editActionConfig]
+  // );
+
+  // const fields = useMemo(()=>{
+  //   if (!(apiToken && appId && apiUrl)) {
+  //     return;
+  //   }
+  //
+  // }, [apiToken, appId, apiUrl]);
+
+  // const [selectedOptions, setSelected] = useState([]);
+  const [isLoading, setLoading] = useState(false);
+  const [options, setOptions] = useState([] as Array<{ label: string; value: string }>);
+  // const { isLoading: isLoadingApplication, application } = useGetApplication({
+  //   http,
+  //   toastNotifications: toasts,
+  //   actionConnector: this,
+  //   id: appId,
+  // });
+
+  // const fieldOptions: EuiSelectOption[] = useMemo(() => {
+  //   return application?.fields.map((f) => ({ value: f.id, text: f.key }));
+  // }, [application]);
+  //
+  // const onFocus = () => {
+  //   this.refs.
+  //   useGetApplication({
+  //     http,
+  //     toastNotifications: toasts,
+  //     actionConnector: this,
+  //     id: appId,
+  //   });
+  // };
+  const abortCtrl = useRef(new AbortController());
+
+  const onSearchChange = useCallback(
+    (searchValue) => {
+      setLoading(true);
+      setOptions([]);
+
+      const didCancel = false;
+
+      const fetchData = async () => {
+        if (!apiToken || !apiUrl || !appId) {
+          setLoading(false);
+          return;
+        }
+
+        abortCtrl.current = new AbortController();
+        setLoading(true);
+
+        try {
+          const res = await getApplication({
+            http,
+            signal: abortCtrl.current.signal,
+            connectorId: action.id,
+            id: appId,
+          });
+          if (!didCancel) {
+            setLoading(false);
+            const application = res.data?.application;
+            if (application?.fields) {
+              const applicationFields = application.fields;
+              const fieldMap = applicationFields.map((f: { id: string; key: string }) => ({
+                value: f.id,
+                label: f.key,
+              }));
+              setOptions(fieldMap);
+            }
+            if (res.status && res.status === 'error') {
+              toasts.addDanger({
+                title: i18n.SW_GET_APPLICATION_API_ERROR(appId),
+                text: `${res.serviceMessage ?? res.message}`,
+              });
+            }
+          }
+        } catch (error) {
+          if (!didCancel) {
+            setLoading(false);
+            toasts.addDanger({
+              title: i18n.SW_GET_APPLICATION_API_ERROR(appId),
+              text: error.message,
+            });
+          }
+        }
+      };
+
+      abortCtrl.current.abort();
+      fetchData();
+    },
+    [toasts, action, appId, apiToken, http, apiUrl]
+  );
+
+  useEffect(() => {
+    // Simulate initial load.
+    onSearchChange('');
+  }, [onSearchChange]);
+
+  return (
+    <Fragment>
```

Comment:
nit -> you do not need fragment
```suggestion
    <>
```